src/LinAlg/load_kkt_mat.m

```diff
@@ -23,9 +26,14 @@
     end
 end
 
-%% form the matrix and fill lower triangle
+%% form the matrix
 M = sparse(rowidx, colidx, vals, m, m);
-M=M+transpose(triu(M,1));
+% if the matrix is known to be symmetric fill, you also need to know whether
+%  A. the lower triangle was saved, in which case use
+% M=M+transpose(tril(M,-1));
+% or
+% B. the upper triangle was saved, in which case use
+% M=M+transpose(triu(M,1));
```

Comment:
how about we do a simple test to check if the input matrix is upper or lower triangular?
For example, use Matlab function `istril()` and `istriu()`. If it is not triangular, return an error.